prime-angular-frontend/src/app/core/services/utils.service.ts

```diff
@@ -77,8 +77,8 @@ export class UtilsService {
    * @description
    * Generic sorting of a JSON object by key.
    */
-  public sortByKey<T>(a: { [key: string]: any }, b: { [key: string]: any }, key: string): SortWeight {
-    return this.sort<T>(a[key], b[key]);
+  public sortByKey<T extends { [key: string]: any }>(key: string): (a: T, b: T) => SortWeight {
```

Comment:
Looks great @johnnyl6.  I made an update to improve the typing for `sortByKey` since this method is  extending from an non-specific object literal instead a specific one so if I sort by a key that doesn't exist on the object there's no protection:

**Was**
```
public sortByKey<T extends { [key: string]: any }>(key: string): (a: T, b: T) => SortWeight { ... }
```

**Now**
```
public sortByKey<T extends { [key: string]: any }>(key: keyof T): (a: T, b: T) => SortWeight { ... }
````